orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -82,16 +82,15 @@ def _combine_input_buffers_initializers(param_names, onnx_input_names, input_inf
 
         elif input_idx >= len(non_none_inputs):
             # Registered buffers are translated to user_input+initializer in ONNX
-            buffer_name, inp = next(named_buffers_iter)
-            assert buffer_name == name, f'Input name {name} expected, but {buffer_name} found!'
+            assert name in buffer_names_dict, f'Buffer name {name} not found!'
+            inp = buffer_names_dict[name]
```

Comment:
Minor performance nit, avoid the double lookup (and be slightly more Pythonic!): 

```suggestion
            try:
              inp = buffer_names_dict[name]
            except KeyError:
              assert False, f'Buffer name {name} not found!'

Reply:
Nice